python/GafferArnoldTest/ArnoldTextureBakeTest.py

```diff
@@ -0,0 +1,473 @@
+##########################################################################
+#
+#  Copyright (c) 2018, Image Engine Design Inc. All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are
+#  met:
+#
+#      * Redistributions of source code must retain the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided with
+#        the distribution.
+#
+#      * Neither the name of John Haddon nor the names of
+#        any other contributors to this software may be used to endorse or
+#        promote products derived from this software without specific prior
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import os
+import inspect
+
+import imath
+
+import IECore
+
+import Gaffer
+import GafferScene
+import GafferSceneTest
+import GafferArnold
+import GafferImage
+import GafferOSL
+import GafferDispatch
+
+class ArnoldTextureBakeTest( GafferSceneTest.SceneTestCase ) :
+
+	class SimpleEdgeDetect( GafferImage.ImageProcessor ):
+
+		def __init__( self, name = "SimpleEdgeDetect" ) :
+			GafferImage.ImageProcessor.__init__( self, name )
+
+			self["HorizTransform"] = GafferImage.ImageTransform()
+			self["HorizTransform"]["in"].setInput( self["in"] )
+			self["HorizTransform"]["transform"]["translate"].setValue( imath.V2f( 1, 0 ) )
+
+			self["HorizDiff"] = GafferImage.Merge()
+			self["HorizDiff"]["in"]["in0"].setInput( self["HorizTransform"]["out"] )
```

Comment:
`["in"][0]`, and other instances on other lines. From the docs for ArrayPlug :

> Currently the element
> names are derived from the name of the first element,
> but this may change in the future. It is strongly
> recommended that ArrayPlug children are only accessed
> through numeric indexing and never via names.

It's an easy mistake to make, especially because we don't refer to them numerically in the UI labels and the serialisation - that's something I need to fix.